gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -48,13 +61,44 @@ def __init__(self, robust_measurement_noise: bool = True) -> None:
         self._max_1dsfm_projection_directions = MAX_PROJECTION_DIRECTIONS
         self._outlier_weight_threshold = OUTLIER_WEIGHT_THRESHOLD
 
+    def __sample_projection_directions(
+        self, 
+        w_i2Ui1_measurements: BinaryMeasurementsUnit3,
+        projection_sampling_method: ProjectionSamplingMethod
+    ) -> List[Unit3]:
+        """Samples projection directions for 1DSfM based on the provided sampling method. 
+        
+        Args:
+            w_i2Ui1_measurements: Unit translation measurements which are input to 1DSfM.
+            projection_sampling_method: ProjectionSamplingMethod to be used for sampling directions. 
+
+        Returns: 
+            List of sampled Unit3 projection directions.
+        """
+        num_measurements = len(w_i2Ui1_measurements)
+
+        if projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_INPUT_MEASUREMENTS:
+            num_samples = min(num_measurements, self._max_1dsfm_projection_directions)
+            sampled_indices = np.random.choice(w_i2Ui1_measurements, num_samples, replace=False)
+            return [w_i2Ui1_measurements[idx].measured() for idx in sampled_indices]
+
+        elif projection_sampling_method == self.ProjectionSamplingMethod.SAMPLE_WITH_INPUT_DENSITY:\
```

Comment:
@akshay-krishnan is there a stray backslash here?

Reply:
ah dang, why did i not see that